builtin/providers/aws/resource_aws_sns_topic_subscription.go

```diff
@@ -178,6 +187,12 @@ func subscribeToSNSTopic(d *schema.ResourceData, snsconn *sns.SNS) (output *sns.
 	protocol := d.Get("protocol").(string)
 	endpoint := d.Get("endpoint").(string)
 	topic_arn := d.Get("topic_arn").(string)
+	endpoint_auto_confirms := d.Get("endpoint_auto_confirms").(bool)
+	confirmation_timeout_in_minutes := time.Duration(d.Get("confirmation_timeout_in_minutes").(int))
```

Comment:
If I don't convert it then I get the follow error
`mismatched types time.Duration and int`
On the other hand, I see your concern that the variable does not make sense
I will change it to 

confirmation_timeout_in_minutes := d.Get("confirmation_timeout_in_minutes").(int) * int(time.Minute)
and use time.Duration(confirmation_timeout_in_minutes)

I have few other options too but I think this makes most sense.